jerry-core/ecma/builtin-objects/ecma-builtin-intrinsic.c

```diff
@@ -64,6 +80,109 @@ ecma_builtin_intrinsic_array_prototype_values (ecma_value_t this_value) /**< thi
+                                         lit_utf8_size_t string_buff_size,
+                                         ecma_value_t radix)
+{
+  return ecma_number_parse_int (string_buff, string_buff_size, radix);
+} /* ecma_builtin_intrinsic_object_parse_int */
+
+/**
+ * The Intrinsic object's 'parseFloat' routine
+ *
+ * See also:
+ *          ECMA-262 v5, 15.1.2.3
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_intrinsic_object_parse_float (const lit_utf8_byte_t *string_buff, /**< routine's first argument's
+                                                                             *   string buffer */
+                                           lit_utf8_size_t string_buff_size) /**< routine's first argument's
+                                                                           *   string buffer's size */
+{
+  return ecma_number_parse_float (string_buff, string_buff_size);
+} /* ecma_builtin_intrinsic_object_parse_float */
+
+/**
+ * Dispatcher of the built-in's routines
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_intrinsic_dispatch_routine (uint16_t builtin_routine_id, /**< built-in wide routine identifier */
+                                         ecma_value_t this_arg, /**< 'this' argument value */
+                                         const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                              *   passed to routine */
+                                         ecma_length_t arguments_number) /**< length of arguments' list */
+{
+  JERRY_UNUSED (arguments_number);
+  /*ecma_string_t *str = ecma_get_string_from_value (arguments_list_p[0]);
+  lit_utf8_size_t str_size = ecma_string_get_size (str);*/
+
+  ecma_value_t routine_arg_1 = arguments_list_p[0];
+  ecma_value_t routine_arg_2 = arguments_list_p[1];
+
+  ecma_string_t *str_p = ecma_op_to_string (routine_arg_1);
+
+  if (JERRY_UNLIKELY (str_p == NULL))
+  {
+    return ECMA_VALUE_ERROR;
+  }
+
+  ecma_value_t ret_value;
+
+  lit_utf8_size_t input_size = ecma_string_get_size (str_p);
+
+  JMEM_DEFINE_LOCAL_ARRAY (input_start_p,
+                           input_size + 1,
+                           lit_utf8_byte_t);
+
+  ecma_string_to_utf8_bytes (str_p, input_start_p, input_size);
+
+  input_start_p[input_size] = LIT_BYTE_NULL;
+
+  switch (builtin_routine_id)
+  {
+    case ECMA_INTRINSIC_PARSE_FLOAT:
+    {
+      ret_value = ecma_builtin_intrinsic_object_parse_float (input_start_p, input_size);
+      break;
+    }
+    case ECMA_INTRINSIC_PARSE_INT:
+    {
+      ret_value = ecma_builtin_intrinsic_object_parse_int (input_start_p, input_size, routine_arg_2);
+      break;
+    }
+    case ECMA_INTRINSIC_ARRAY_PROTOTYPE_VALUES:
+    {
+      ret_value = ecma_builtin_intrinsic_array_prototype_values (this_arg);
```

Comment:
Tostring , and get string buffer operation should not be applied for this method.